src/course-unit/SubsectionUnitRedirect.tsx

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from 'react';
+import { LoadingSpinner } from '@src/generic/Loading';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useCourseItemData } from '../course-outline/data/apiHooks';
+
+const SubsectionUnitRedirect = ({ courseId }: { courseId: string }) => {
+  let { subsectionId } = useParams();
+  // if the call is made via the click on breadcrumbs the re won't be courseId available
+  // in such cases the page should redirect to the 1st unit of he subsection
+  const navigate = useNavigate();
+  const { data: courseItemData } = useCourseItemData(subsectionId);
+  let firstUnitId = courseItemData?.childInfo?.children?.[0]?.id;
+  let url: string;
+
+  useEffect(() => {
+    if (firstUnitId) {
+      firstUnitId = encodeURIComponent(firstUnitId);
+      url = `/course/${courseId}/container/${firstUnitId}`;
+      navigate(url, { replace: true });
+    }
+    // if no unit then navigate to the subsection outline
+    if (courseItemData?.childInfo?.children?.length === 0 && subsectionId) {
+      subsectionId = encodeURIComponent(subsectionId);
+      url = `/course/${courseId}?show=${subsectionId}`;
+      navigate(url, { replace: true });
+    }
+  }, [firstUnitId, courseItemData, subsectionId]);
+  return <LoadingSpinner />;
```

Comment:
If you want, instead of `useEffect`, you can return a `<Navigate` component, which is more declarative:

```tsx
return <Navigate replace to={`/course/${courseId}/container/${firstUnitId}`} />
```